pkg/controller/job/job_controller.go

```diff
@@ -1049,6 +1045,7 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 			//
 			// The job will be synced again because the Job status and Pod updates
 			// will put the Job back to the work queue.
+			reachedMaxUncountedPods = true
```

Comment:
Side question: I'm wondering why `newSucceededIndexes` also contributes to the counting here since it does not expand the field. Thus, we may be blocking in some scenarios marking of the job as finished too early.

Reply:
yes, but there are so many API calls already. There will be a next sync anyways.

Reply:
sgtm